central/metadata/service/service_impl.go

```diff
@@ -133,3 +138,31 @@ func (s *serviceImpl) TLSChallenge(ctx context.Context, req *v1.TLSChallengeRequ
 
 	return resp, nil
 }
+
+// GetDatabaseStatus returns the database status for Rox.
+func (s *serviceImpl) GetDatabaseStatus(ctx context.Context, _ *v1.Empty) (*v1.DatabaseStatus, error) {
+	dbStatus := &v1.DatabaseStatus{
+		DatabaseAvailable: true,
+	}
+
+	dbType := "RocksDB"
+	var dbVersion string
+	if env.PostgresDatastoreEnabled.BooleanSetting() {
+		dbType = "PostgresDB"
+		if err := s.db.Ping(ctx); err != nil {
+			dbStatus.DatabaseAvailable = false
+			log.Warn("central is unable to communicate with the database.")
+			return dbStatus, nil
+		}
+
+		dbVersion = globaldb.GetPostgresVersion(ctx, s.db)
+	}
+
+	// Only return the database type and version to logged in users, not anonymous users.
+	if authn.IdentityFromContextOrNil(ctx) != nil {
```

Comment:
Why do we want to provide the database availability for anonymous users?
And why not expose the auth failure so the customer can fix it?

Reply:
@c-du Because if central cannot talk to the database then UI cannot provide the auth providers on the login page.  So allowing for that flag to be returned when the user is not authenticated can let them know earlier that the database is the issue.  this function is similar to how the /metadata endpoint works.

Reply:
Yes, that is right.